c/extras/py_ftrl.cc

```diff
@@ -212,198 +328,247 @@ Returns
-    throw ValueError() << "FTRL model frame must have " << fm->get_d() << " rows,"
-                       << "and 2 columns, whereas your frame has " << dt_model_in->nrows
-                       << " rows and " << dt_model_in->ncols << " columns";
+  if (dt_model_in->nrows != ft->get_d() || dt_model_in->ncols != 2) {
+    throw ValueError() << "FTRL model frame must have " << ft->get_d()
+                       << " rows, and 2 columns, whereas your frame has "
+                       << dt_model_in->nrows << " rows and "
+                       << dt_model_in->ncols << " columns";
+
   }
 
-  if (model_cols_in != FtrlModel::model_cols) {
-    throw ValueError() << "FTRL model frame must have columns named `z` and `n`,"
-                       << "whereas your frame has the following column names `" << model_cols_in[0]
+  if (model_cols_in != Ftrl::model_cols) {
+    throw ValueError() << "FTRL model frame must have columns named `z` and "
+                       << "`n`, whereas your frame has the following column "
+                       << "names: `" << model_cols_in[0]
                        << "` and `" << model_cols_in[1] << "`";
   }
 
   if (dt_model_in->columns[0]->stype() != SType::FLOAT64 ||
     dt_model_in->columns[1]->stype() != SType::FLOAT64) {
-    throw ValueError() << "FTRL model frame must have both column types as `float64`, "
-                       << "whereas your frame has the following column types: `"
+    throw ValueError() << "FTRL model frame must have both column types as "
+                       << "`float64`, whereas your frame has the following "
+                       << "column types: `"
                        << dt_model_in->columns[0]->stype()
                        << "` and `" << dt_model_in->columns[1]->stype() << "`";
   }
 
-  fm->set_model(dt_model_in);
+  ft->set_model(dt_model_in);
 }
 
 
 /*
 *  All other getters and setters.
 */
-oobj Ftrl::get_a() const {
-  return py::ofloat(fm->get_a());
+oobj PyFtrl::get_colnames_hashes() const {
+  if (ft->is_trained()) {
+    size_t n_features = ft->get_n_features();
+    py::otuple py_colnames_hashes(n_features);
+    std::vector<uint64_t> colnames_hashes = ft->get_colnames_hashes();
+    for (size_t i = 0; i < n_features; ++i) {
+      size_t h = static_cast<size_t>(colnames_hashes[i]);
+      py_colnames_hashes.set(i, py::oint(h));
+    }
+    return std::move(py_colnames_hashes);
+  } else {
+    return py::None();
+  }
+
 }
 
 
-oobj Ftrl::get_b() const {
-  return py::ofloat(fm->get_b());
+oobj PyFtrl::get_params() const {
+  py::onamedtuple params(params_nttype);
+  params.set(0, get_alpha());
+  params.set(1, get_beta());
+  params.set(2, get_lambda1());
+  params.set(3, get_lambda2());
+  params.set(4, get_d());
+  params.set(5, get_n_epochs());
+  params.set(6, get_inter());
+  return std::move(params);
 }
 
 
-oobj Ftrl::get_l1() const {
-  return py::ofloat(fm->get_l1());
+oobj PyFtrl::get_default_params() const {
```

Comment:
Also not needed: same as `Ftrl().params`

Reply:
Yes, this one can be removed.